avocado/utils/asset.py

```diff
@@ -476,6 +477,47 @@ def get_asset_by_name(cls, name, cache_dirs, expire=None, asset_hash=None):
 
         raise OSError("File %s not found in the cache." % name)
 
+    @classmethod
+    def get_assets_by_size(cls, size_filter, cache_dirs):
+        """Return a list of all assets in cache based on its size in MB.
+
+        :param size_filter: a string with a filter (comparison operator +
+        value). Ex ">20", "<=200". Supported operators: ==, <, >, <=, >=.
+        :param cache_dirs: list of directories to use during the search.
+        """
+        operators = {'==': operator.eq,
+                     '<': operator.lt,
+                     '>': operator.gt,
+                     '<=': operator.le,
+                     '>=': operator.ge}
+        try:
+            op = re.match('^(\\D+)(\\d+)$', size_filter).group(1)
+            value = int(re.match('^(\\D+)(\\d+)$', size_filter).group(2))
+        except (AttributeError, ValueError):
+            msg = ("Invalid syntax. You need to pass an comparison operatator",
+                   " and a value. Ex: '>=200'")
+            raise Exception(msg)
+
+        try:
+            method = operators[op]
+        except KeyError:
+            msg = ("Operator not supported. Currented valid values are:",
+                   "==, <, >, <=, >=")
```

Comment:
This list of supported operators should come from the `operators` dict itself to avoid future inconsistencies and bit rot.